### ISSUE TITLE
Where is the PSD Background Layer

### ISSUE THREAD

**Issue by slgaficon** [ImageMagick/ImageMagick#2586]
In IM 6.x any conversion to PSD format resulted in a PSD where the original image is found in the default background layer.
Using IM 7.x  a custom layer "L1" is created an the original image is placed there. Using command line option -layers flatten does not have any effect.
Opening the PSD in Photoshop, flattening there results in a PSD file with roughly half the size as IM 7 created. This is a lot of unnecessary disk space wasted concerning you have some million images.
So, how can I convert to a PSD with everything placed in the default Background layer?

**Comment by urban-warrior** [comment]
We need to reproduce the problem and compare the results betweem IM 6 & 7.  Once we identify the issue, we can offer feedback or a solution.  Can you post a URL to an image we can download and the command line you are using?

**Comment by fmw42** [comment]
I have done the following and get the same result with IM 6.9.11.29 and IM 7.010.29. Both put the image into the only layer which is labeled L1.

`convert lena.png lena6.psd`

and


`magick lena.png lena7.psd`


![Screen Shot 2020-09-16 at 3 03 43 PM](https://user-images.githubusercontent.com/13458218/93397246-0a1bce80-f82e-11ea-8051-2cde276f8a38.png)

**Comment by slgaficon** [comment]
convert command:
convert -units PixelsPerInch "2020-09-17\19_Kirschen_05_4c_DE Test.jpg[0]" ( ) -units PixelsPerInch -depth 8 -density 300 -unsharp 0.0x0.0+0.0+0.0 +profile icc +profile 8bim -profile "O:\Volume0\db-main\opastool\funcd\tools\color_convert\devicelink_profiles\Adobe RGB (1998)_ISOcoated_v2_eci.icc" -set profile "O:/Volume0/db-main/opastool/funcd/tools/color_convert/icc_profiles/ISOcoated_v2_eci.icc" "D:\kirschen.psd"

Using ImageMagick-6.8.0-Q16 result in background layer:
![image](https://user-images.githubusercontent.com/47604663/93454369-317ca500-f8db-11ea-90d8-7569b56fab24.png)

Using IM 6.9 or 7.x results in:
![image](https://user-images.githubusercontent.com/47604663/93454451-54a75480-f8db-11ea-8dd1-7d997c1c7042.png)

Regarding dis space:
![image](https://user-images.githubusercontent.com/47604663/93454558-73a5e680-f8db-11ea-905d-e60d4d626d3e.png)
The first PSD has been converted with IM 6.8

**Comment by slgaficon** [comment]
To reproduce one can use any JPG image and a simple "convert x.jpg y.psd" command

**Comment by fmw42** [comment]
It still does not happen that way for me even with jpg. Both show L1. Using IM 6.9.11.29 and 7.0.10.29 Q16 Mac OSX Sierra

`convert lena.jpg lena6.psd`

`magick lena.jpg lena7.psd`

![Screen Shot 2020-09-17 at 9 23 09 AM](https://user-images.githubusercontent.com/13458218/93499094-70a0fb00-f8c7-11ea-8828-5ce9aec48c9f.png)

**Comment by slgaficon** [comment]
You are right. The difference will show up when you use ImageMagick-6.8.0-Q16!

**Comment by slgaficon** [comment]
Any news? Why did the behavior change?

**Comment by slgaficon** [comment]
Another thought: Should command line option "flatten" reduce all layers to the background layer in case of a PDF file?
This is at least what the documentation suggests.
So, if flattening would be equivalent to reducing PSD content to the default background layer, our requirement would be fulfilled.
Is this an option?

**Comment by meredith-jannear** [comment]
Any update on this?

**Comment by dlemstra** [comment]
This is happening because we always write layer information now. Earlier version of ImageMagick would not always write layer information. We could introduce a define that can be used to disable writing layer information. Maybe `-define psd:write-layers=false` @slgaficon?

**Comment by slgaficon** [comment]
Would be a great help to have this option. PSD files increase in size dramatically since the change.

**Comment by dlemstra** [comment]
Starting with the next version of ImageMagick 7 you will be able to use `psd:write-layers=false` to disable writing of the layers. This new feature will not be backported to ImageMagick 6.
